### PR TITLE
add kernel for cross function

### PR DESCRIPTION
**Pull request by KsanaKozlova**

### CODE REVIEW

File: dpnp/backend/include/dpnp_iface.hpp [hunk 1]
```diff
@@ -156,6 +156,22 @@ INP_DLLEXPORT void dpnp_elemwise_absolute_c(void* array1_in, void* result1, size
 template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
 INP_DLLEXPORT void dpnp_dot_c(void* array1, void* array2, void* result1, size_t size);
 
+/**
+ * @ingroup BACKEND_API
+ * @brief Custom implementation of dot function
+ *
+ * @param [in]  array1  Input array.
+ *
+ * @param [in]  array2  Input array.
```
Comment (densmirn) [inline dpnp/backend/include/dpnp_iface.hpp:165]:
```suggestion
 * @brief Custom implementation of cross function
 *
 * @param [in]  array1_in  First input array.
 *
 * @param [in]  array2_in  Second input array.
```

File: dpnp/backend/kernels/dpnp_krnl_mathematical.cpp [hunk 2]
```diff
@@ -125,6 +125,30 @@ void dpnp_floor_divide_c(void* array1_in, void* array2_in, void* result1, size_t
...
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+void dpnp_cross_c(void* array1_in, void* array2_in, void* result1, size_t size)
+{
+
+    _DataType_input1* array1 = reinterpret_cast<_DataType_input1*>(array1_in);
+    _DataType_input2* array2 = reinterpret_cast<_DataType_input2*>(array2_in);
+    _DataType_output* result = reinterpret_cast<_DataType_output*>(result1);
+
+    _DataType_output cur_res = array1[1] * array2[2] - array1[2] * array2[1];
+    result[0] = cur_res;
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_mathematical.cpp:140]: Why not combine 2 lines to a single one here and below?

File: dpnp/backend/include/dpnp_iface.hpp (hunk 1 above)
Comment (samir-nasibli) [reply dpnp/backend/include/dpnp_iface.hpp:165]: Please remove blank lines between param descriptions. See example: https://github.com/IntelPython/dpnp/blob/master/dpnp/backend/include/dpnp_iface.hpp#L151:L154

File: dpnp/backend/kernels/dpnp_krnl_mathematical.cpp [hunk 3]
```diff
@@ -244,6 +263,23 @@ void func_map_init_mathematical(func_map_t& fmap)
     fmap[DPNPFuncName::DPNP_FN_FLOOR_DIVIDE][eft_DBL][eft_DBL] = {eft_DBL,
                                                                   (void*)dpnp_floor_divide_c<double, double, double>};
 
+    fmap[DPNPFuncName::DPNP_FN_CROSS][eft_INT][eft_INT] = {eft_INT, (void*)dpnp_cross_c<int, int, int>};
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_mathematical.cpp:266]: Please insert the mapping in alphabetical order by name of the function.

**Review by densmirn — APPROVED** [review]

File: dpnp/backend/kernels/dpnp_krnl_mathematical.cpp [hunk 4]
```diff
@@ -83,6 +83,25 @@ template void dpnp_elemwise_absolute_c<float>(void* array1_in, void* result1, si
...
+{
+    _DataType_input1* array1 = reinterpret_cast<_DataType_input1*>(array1_in);
+    _DataType_input2* array2 = reinterpret_cast<_DataType_input2*>(array2_in);
+    _DataType_output* result = reinterpret_cast<_DataType_output*>(result1);
+
+    result[0] = array1[1] * array2[2] - array1[2] * array2[1];
+
+    result[1] = array1[2] * array2[0] - array1[0] * array2[2];
+
+    result[2] = array1[0] * array2[1] - array1[1] * array2[0];
```
Comment (Alexander-Makaryev) [inline dpnp/backend/kernels/dpnp_krnl_mathematical.cpp:100]: Is it really common enough case for this function?
This looks like a very special case.

File: dpnp/backend/kernels/dpnp_krnl_mathematical.cpp [hunk 5]
```diff
@@ -83,6 +83,25 @@ template void dpnp_elemwise_absolute_c<float>(void* array1_in, void* result1, si
...
+
+    return;
+}
```
Comment (Alexander-Makaryev) [inline dpnp/backend/kernels/dpnp_krnl_mathematical.cpp:103]: This is the complete native function for better understanding of my previous comment.

File: dpnp/backend/kernels/dpnp_krnl_mathematical.cpp [hunk 6]
```diff
@@ -83,6 +83,25 @@ template void dpnp_elemwise_absolute_c<float>(void* array1_in, void* result1, si
 template void dpnp_elemwise_absolute_c<long>(void* array1_in, void* result1, size_t size);
 template void dpnp_elemwise_absolute_c<int>(void* array1_in, void* result1, size_t size);
 
+template <typename _KernelNameSpecialization1, typename _KernelNameSpecialization2, typename _KernelNameSpecialization3>
+class dpnp_cross_c_kernel;
+
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+void dpnp_cross_c(void* array1_in, void* array2_in, void* result1, size_t size)
```
Comment (samir-nasibli) [inline dpnp/backend/kernels/dpnp_krnl_mathematical.cpp:90]: There is unused param `size`. Is there any common tests ?

**Comment by samir-nasibli** [comment]
@shssf please provide some comment before merging this kind of PRs. Feature is not completely implemented.

**Pull request merged**